mavlink/modules/flight_controller.py

```diff
@@ -95,3 +95,21 @@ def get_home_location(self, timeout: float) -> "tuple[bool, drone_odometry.Drone
             return False, None
 
         return True, location
+
+    def drone_travelling_to_final_waypoint(self) -> "tuple[bool, bool | None]":
+        """
+        Returns whether the drone is travelling to the final waypoint in the mission.
+        """
+        waypoint_count = self.drone.commands.count
+        current_waypoint = self.drone.commands.next
+
+        if waypoint_count < 0 or current_waypoint < 0:
+            return False, None
+
+        if waypoint_count == 0:
+            return True, False
+        elif waypoint_count > 0:
```

Comment:
Remove unnecessary else (the previous is a return, and the previous condition checks guarantee that `waypoint_count > 0` .